### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Refactors swap/trade component
* Adds new settings style
* Moves slippage provider up in the tree
* Adds new `useBalance` hook (for now only used in `Swap` but soon to be used in other places)
* Adds selected token provider to prepare for using paths for input/output token determination
* Updates safary to be loaded via [Script](https://nextjs.org/docs/app/building-your-application/optimizing/scripts)
* Slowly adds `bigint` from viem to the project (so there might be some conversions between `bigint` and `BigNumber` here and there until everything's converted)
* Removes obsolete prettier + eslint ignore (contained folders are ignored by default)
* Removes sentry

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: .prettierignore
Comment (0xonramp) [inline .prettierignore:1]: Not used anymore?

File: src/lib/utils/index.ts [hunk 1]
```diff
@@ -2,9 +2,28 @@ import { BigNumber } from 'ethers'
...
 import { formatUnits, parseUnits } from '@ethersproject/units'
 
+export function isSameAddress(address1: string, address2: string): boolean {
+  return address1.toLowerCase() === address2.toLowerCase()
+}
+
 export const selectLatestMarketData = (marketData?: number[][]) =>
   marketData?.[marketData.length - 1]?.[1] || 0
 
+export function shortenAddress(
```
Comment (0xonramp) [inline src/lib/utils/index.ts:12]: 🔥

File: src/lib/hooks/use-balance.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    return await this.publicClient.readContract({
+      address: token as Address,
+      abi: ERC20_ABI,
+      functionName: 'balanceOf',
+      args: [address as Address],
+    })
+  }
+
+  async getNativeBalance(address: string) {
+    return await this.publicClient.getBalance({
```
Comment (0xonramp) [inline src/lib/hooks/use-balance.ts:20]: nit: can remove these two `await`s

File: src/lib/hooks/use-best-quote/flashmint.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      // We don't want this function to fail for estimates here.
+      // A default will be returned if the tx would fail.
+      const canFail = false
+      tx.chainId = 1
+      tx.from = from
+      tx.gasLimit = defaultGasEstimate
+      const gasEstimate = await gasEstimatooor.estimate(transaction, canFail)
+      const gasCosts = gasEstimate.mul(gasPrice)
+      const gasCostsInUsd = getGasCostsInUsd(gasCosts, nativeTokenPrice)
+      transaction.gasLimit = gasEstimate
```
Comment (0xonramp) [inline src/lib/hooks/use-best-quote/flashmint.ts:76]: Should `tx` and `transaction` both be updated in this block, or just `transaction`?

Comment (janndriessen) [reply src/lib/hooks/use-best-quote/flashmint.ts:76]: Good point, will def revisit this part.

**Pull request merged**